package.json

```diff
@@ -46,60 +49,60 @@
     "gulp-rename": "^1.2.2",
     "gulp-replace": "^0.6.1",
     "gulp-sass": "^3.1.0",
-    "gulp-sourcemaps": "^2.6.0",
+    "gulp-sourcemaps": "^2.6.1",
     "gulp-strip-css-comments": "^1.2.0",
-    "gulp-stylelint": "^4.0.0",
+    "gulp-stylelint": "^5.0.0",
     "gulp-tslint": "^8.1.2",
-    "gulp-typescript": "^3.2.1",
+    "gulp-typescript": "^3.2.2",
     "gulp-util": "^3.0.8",
     "highlights": "^3.0.1",
     "http-server": "^0.10.0",
-    "istanbul-instrumenter-loader": "^0.2.0",
-    "json-loader": "^0.5.4",
-    "karma": "^1.7.0",
+    "istanbul-instrumenter-loader": "^3.0.0",
+    "json-loader": "^0.5.7",
+    "karma": "^1.7.1",
     "karma-chai": "^0.1.0",
     "karma-chrome-launcher": "^2.2.0",
     "karma-coverage": "^1.1.1",
     "karma-firefox-launcher": "^1.0.1",
     "karma-mocha": "^1.3.0",
-    "karma-mocha-reporter": "^2.2.3",
+    "karma-mocha-reporter": "^2.2.5",
     "karma-phantomjs-launcher": "^1.0.4",
-    "karma-phantomjs-shim": "^1.4.0",
+    "karma-phantomjs-shim": "^1.5.0",
     "karma-sinon": "^1.0.5",
     "karma-sourcemap-loader": "^0.3.7",
-    "karma-webpack": "^2.0.4",
+    "karma-webpack": "^2.0.5",
     "language-less": "github:atom/language-less",
     "language-typescript": "github:giladgray/language-typescript#10.1.15",
-    "lerna": "^2.0.0",
+    "lerna": "^2.4.0",
     "lodash": "^4.17.4",
     "marked": "^0.3.6",
     "merge-stream": "^1.0.1",
-    "mocha": "^3.4.2",
+    "mocha": "^4.0.1",
     "node-sass-package-importer": "^3.0.4",
-    "npm-run-all": "^4.0.2",
-    "phantomjs-prebuilt": "^2.1.14",
-    "postcss-import": "^10.0.0",
-    "postcss-url": "^7.1.1",
-    "prettier": "^1.7.0",
+    "npm-run-all": "^4.1.1",
+    "phantomjs-prebuilt": "^2.1.15",
+    "postcss-import": "^11.0.0",
+    "postcss-url": "^7.1.2",
+    "prettier": "^1.7.4",
     "react": "^15.5.1",
     "react-dom": "^15.5.1",
     "react-test-renderer": "^15.5.4",
-    "run-sequence": "^2.0.0",
+    "run-sequence": "^2.2.0",
     "sinon": "^1.17.6",
     "sorted-object": "^2.0.1",
-    "source-map-loader": "^0.2.1",
+    "source-map-loader": "^0.2.2",
     "string-replace-loader": "^1.3.0",
-    "stylelint": "^8.0.0",
+    "stylelint": "^8.2.0",
     "stylelint-config-palantir": "^2.1.0",
     "stylelint-config-standard": "^17.0.0",
-    "stylelint-scss": "^1.5.1",
-    "ts-loader": "^1.3.3",
-    "ts-quick-docs": "^0.5.1",
+    "stylelint-scss": "^2.1.0",
+    "ts-loader": "^3.0.3",
+    "ts-quick-docs": "^0.5.3",
     "tslint": "^5.7.0",
-    "tslint-config-prettier": "^1.5.0",
-    "tslint-plugin-prettier": "^1.1.0",
+    "tslint-config-prettier": "^1.6.0",
+    "tslint-plugin-prettier": "^1.3.0",
     "tslint-react": "^3.2.0",
-    "typescript": "~2.2.1",
+    "typescript": "~2.4.2",
```

Comment:
😱 hooray! is this a safe change for .d.ts compatibility?

Reply:
Should be safe since we're not using new features, but there are no guarantees. I would just highlight the small chance of breaking syntax in the release notes and file it as a low priority issue to fix in the future. I don't want to wait until the next major version to upgrade typescript.